src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework.UnitTests/Mocks/TestMethodHandlers.cs

```diff
@@ -149,3 +149,35 @@ public Task<string> HandleRequestAsync(int request, TestRequestContext context,
         throw new NotImplementedException();
     }
 }
+
+[LanguageServerEndpoint(Name)]
+internal class TestDefaultLanguageHandler : IRequestHandler<int, string, TestRequestContext>
+{
+    public const string Name = "Language";
+    public static readonly IMethodHandler Instance = new TestDefaultLanguageHandler();
+
+    public bool MutatesSolutionState => true;
+    public static Type RequestType = typeof(int);
+    public static Type ResponseType = typeof(string);
+    public static RequestHandlerMetadata Metadata = new(Name, RequestType, ResponseType);
+
+    public Task<string> HandleRequestAsync(int request, TestRequestContext context, CancellationToken cancellationToken)
+        => Task.FromResult(string.Empty);
+}
+
+[LanguageServerEndpoint(Name, Language)]
+internal class TestXamlLanguageHandler : IRequestHandler<int, string, TestRequestContext>
```

Comment:
i don't really get what tehse test handlers are doing.

Reply:
They don't really do anything. They are just there to test that the correct handler for a given language is returned.